code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -107,6 +108,22 @@ GLOBAL_LIST_EMPTY(PDAs)
 		return attack_self(M)
 	return
 
+/obj/item/device/pda/ui_action_click()

```

Comment:
check normal actions then, it's no different.
```ui_action_click()``` is a shitty hold-over from before action datums, and it needs to die.